### PR TITLE
Update: Add make generate and make openapi3gen part of CI workflow

### PR DESCRIPTION
**Pull request by ericrpowers**
#### Description
Noticed today during the demo that we had a desync between our open API files and master gaia changes. To avoid this going forward, `openapi3gen` is now part of the CI workflow and we have a new command that everyone can run to generate all necessary files with:
```
make generate
```
This generic command will now house all necessary generation and formatting needed and can be modified in the future without needing to learn yet another new `make` command.

> Note: I also went ahead and bumped our support for golang v1.18 (and deps) and dropped 1.15 (keeping with support of the 3 more recent releases).

### CODE REVIEW

**Review by malkhamis-panw — DISMISSED** [review]

**Review by t00f — APPROVED** [review]
LGTM, let's have @malkhamis-panw having a look too

File: .github/workflows/build-go.yaml [hunk 1]
```diff
@@ -19,40 +19,26 @@ jobs:
...
         run: |
           go version
-          go get github.com/securego/gosec/cmd/gosec
-          #go get golang.org/x/tools/cmd/cover
-          go get github.com/axw/gocov/gocov
-          go get github.com/AlekSi/gocov-xml
-          go get go.aporeto.io/elemental/cmd/elegen
-          go get go.aporeto.io/regolithe/cmd/rego
-          git checkout go.mod go.sum
+          go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
```
Comment (malkhamis-panw) [inline .github/workflows/build-go.yaml:35]: let's keep our heads up for when they decide to introduce a new feature that breaks the build out of the blue.. this happened to me in the past such that I preferred not to pin to `latest`

**Review by malkhamis-panw — APPROVED** [review]
just one minor comment to be aware of

File: .github/workflows/build-go.yaml (hunk 1 above)
Comment (ericrpowers) [reply .github/workflows/build-go.yaml:35]: FWIW, we were always pointed to latest for anything not v1.15. I just moved this to the main setup after ridding us of golang v1.15.

**Pull request merged**